modules/ObjectControl/src/objectcontrol.cpp

```diff
@@ -875,6 +875,12 @@ void ObjectControl::returnTrajectoryCallback(const rclcpp::Client<atos_interface
 void ObjectControl::setObjectTrajectory(uint32_t id){
 	// Check if object is in resetting "state". If so, offer return trajectory instead of normal trajectory
 	if (this->isResetting) {
+		using namespace std::chrono_literals;
+		if (!this->returnTrajectoryClient->wait_for_service(1s))
+		{
+			RCLCPP_ERROR(get_logger(), "The return trajectory service is not available. Try again later.");
+			return;
+		}
```

Comment:
Can this lead to issues since wait_for_service is blocking?

Reply:
Not really since I designed it to only look once if the service is available (if-statement instead of while-statement). If it doesn't find the service after just one look, it will just return.